pkg/catalog/manager/traverse.go

```diff
@@ -138,7 +177,43 @@ func traverseFiles(repoPath string, catalog *v3.Catalog) ([]v3.Template, []strin
 			toDeleteChart = append(toDeleteChart, fmt.Sprintf("%s-%s", catalog.Name, chart))
 		}
 	}
-	return templates, toDeleteChart, newHelmVersionCommits, nil, nil
+	// delete non-existing templates
+	for _, toDelete := range toDeleteChart {
```

Comment:
The loop body is a bit big. Can we extract some pieces into separate methods? Like deleteTemplate, etc